platform/mbed_rtc_time.h

```diff
@@ -27,6 +27,14 @@
 extern "C" {
 #endif
 
+/* Timeval definition for non GCC_ARM toolchains */
+#if !defined(__GNUC__) || defined(__CC_ARM)
+struct timeval {
+    time_t tv_sec;
+    time_t tv_usec;
```

Comment:
I find it confusing to use `time_t` as a type for `tv_usec` as `time_t` is quite coupled with a second granularity. Linux (and maybe other Unix/Posix flavours) use the `suseconds_t` type for `tv_usec`.

Reply:
Yep. Even if we don't define POSIX's `suseconds_t`, it is supposed to be a signed integer, while `time_t` is probably unsigned, or could even be 64-bit. `long` would do.